In [ ]:
from  scipy import stats
import numpy as np

In [ ]:
ls #Displays flies in current working folder

sample_data/


In [ ]:
import os
os.getcwd()

'/content'

Part (a): 99% Confidence Interval Using Sample Standard Deviation


1.Calculate the sample mean and standard deviation.

2.Use the t-distribution to compute the confidence interval since we are using the sample standard deviation.


Step 1: State the problem clearly.
We are asked to build a 99% confidence interval for the mean number of characters printed before failure for a print-head. The sample size is 15, and we are assuming the sample standard deviation is representative of the population. Since the sample size is small (n = 15), we will use the t-distribution for this task.

Step 2: Gather the necessary data.
The data provided is: 1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29 (in millions of characters).

We need to calculate:

length of data(n)

The sample mean (x̅)

The sample standard deviation (s)

In [ ]:
#The durability of each print-head (in millions of characters)
data=np.array([1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29])
n=len(data)# n=sample size
sample_mean=np.mean(data)#x̅=smaple mean
sample_std=np.std(data)#s-smaple standard deviation
confidence_level=0.99 #1-alpha=confidence level


Step 3: Calculate the confidence interval.
The confidence interval is calculated as:

In [ ]:
con_std_sample=stats.t.interval(confidence_level, df=n-1, loc=sample_mean, scale=sample_std/np.sqrt(n))
print(f"Part (a) - 99% Confidence Interval (using sample standard deviation",np.round(con_std_sample,4))

Part (a) - 99% Confidence Interval (using sample standard deviation [1.0952 1.3821]


Explaination-since population standard deviation is unkonwn and the smaple size is small , we have to use t-distribution for confidence interval.The t-distribution is more appropriate than the normal distribution when dealing with small sample sizes and when the population standard deviation is not known. The t-distribution accounts for the increased variability expected in small samples.

Part b: Build 99% Confidence Interval Using Known Population Standard Deviation

Step 1: State the problem.
Now, we're asked to construct the 99% confidence interval for the mean number of characters printed before failure, but with the assumption that the population standard deviation is 0.2 million characters. In this case, we will use the z-distribution because the population standard deviation is known.

Step 2: Gather the necessary data.

length of data(n)

The sample mean (x̅)

population standard deviation = 0.2


Step 3: Calculate the confidence interval.

In [ ]:
population_std = 0.2 # Given population standard deviation
n=len(data)# n=sample size
sample_mean=np.mean(data)#x̅=smaple mean
confidence_level=0.99 #1-alpha=confidence level
con_std_population=stats.norm.interval(confidence_level, loc=sample_mean, scale=population_std/np.sqrt(n))
print(f"Part (b) - 99% Confidence Interval (using population standard deviation):",np.round(con_std_population,4))

Part (b) - 99% Confidence Interval (using population standard deviation): [1.1057 1.3717]


Summary of Results:
Using sample standard deviation (t-distribution): 99% confidence interval = [1.0952 1.3821] million characters.
Using known population standard deviation (z-distribution): 99% confidence interval = [1.1057 1.3717] million characters.